test/integration/puppetlabs/services/master/environment_classes_int_test.clj

```diff
@@ -92,9 +94,47 @@
   [response]
   (-> response :body cheshire/parse-string))
 
-(deftest ^:integration environment-classes-integration-test
+(deftest ^:integration environment-classes-integration-cache-disabled-test
+  (testing "when environment classes cache is disabled for a class request"
+    (bootstrap/with-puppetserver-running-with-config
+     app
+     (-> {:jruby-puppet {:max-active-instances 1}}
+         (bootstrap/load-dev-config-with-overrides)
+         (ks/dissoc-in [:puppetserver
+                        :environment-class-cache-enabled]))
```

Comment:
For any test that expects a specific value to be used for a setting - in this case that the `environment-class-cache-enabled` setting not be present in the configuration but the default behavior of "false" be used - I think the setting should be explicitly set within the test itself.  It would seem bad if in the future someone had changed the setting in the dev sample config file without understanding the implicit behavior that this test was relying upon but having the test still pass despite an underlying problem in the implementation being present.  Making the setting explicit in the test that depends upon its value seems like the best safeguard to me.

Note that I followed this line of reasoning with the corresponding `environment-classes-integration-cache-enabled-test` as well where I explicitly set a value of `true` for the setting even though the dev sample config used the same value.